lib/spack/spack/build_systems/cmake.py

```diff
@@ -145,6 +145,10 @@ def _std_args(pkg):
         args.append('-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=FALSE')
         rpaths = ':'.join(spack.build_environment.get_rpaths(pkg))
         args.append('-DCMAKE_INSTALL_RPATH:STRING={0}'.format(rpaths))
+        # CMake's find_package() looks in CMAKE_PREFIX_PATH first, help CMake
+        # to find immediate link dependencies in right places:
+        deps = [d.prefix for d in pkg.spec.dependencies(deptype='link')]
+        args.append('-DCMAKE_PREFIX_PATH:STRING={0}'.format(';'.join(deps)))
```

Comment:
@mathstuf does this look reasonable to you?

Reply:
As long as `;` can't be generated in prefixes :) . If Spack doesn't generate those itself, I'd say it's a "just rename your directory" solution.

Reply:
> As long as ; can't be generated in prefixes

hopefully `;` does not appear in prefixes.

Reply:
I think using:
```python
 deps = [d.prefix for d in pkg.spec.dependencies(deptype=('build', 'link'))]
```
is more correct (think for instance at using tools like `bison` or similar during build). @mathstuf @adamjstewart @tgamblin What do you think?